olive/engine/engine.py

```diff
@@ -879,7 +890,7 @@ def _evaluate_model(
         return signal
 
     def _get_top_ranked_nodes(self, metrics: List[Metric], footprint: Footprint, k: int) -> List[FootprintNode]:
-        metric_priority = [metric.name for metric in sorted(metrics, key=lambda x: x.priority_rank)]
+        metric_priority = [metric.name for metric in sorted(metrics, key=lambda x: x.priority)]
```

Comment:
There is no `priority` attribute for Metric, should we use the subtype's priority instead here?

Reply:
Updated to use sorted objective_dict.